unlambda panic's for variadic args

Golang version 1.15.2
Gocritic version v0.5.0

We found that gocritic's check unlambda panic's when dealing with function signatures that have variadic args

Our simple example is 
```golang
func TestSomething(t *testing.T) {
	fn :=  func(t assert.TestingT, e interface{}, a interface{}, _ ...interface{}) bool {
		return assert.Equal(t, e, a)
	}

	fn(t, "apples", "oranges")
}
```

when we run `gocritic check` on the package we get

```
unlambda: error: runtime error: index out of range [3] with length 3
panic: runtime error: index out of range [3] with length 3 [recovered]
        panic: runtime error: index out of range [3] with length 3

goroutine 2558 [running]:
github.com/go-critic/go-critic/framework/lintmain/internal/check.(*program).checkFile.func1.1(0xc0005bcc10, 0xc000f35b00)
        /Users/aperson/pkg/mod/github.com/go-critic/go-critic@v0.5.2/framework/lintmain/internal/check/check.go:142 +0x15b
panic(0x43dde40, 0xc0007b22e0)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/panic.go:969 +0x166
github.com/go-critic/go-critic/checkers.(*unlambdaChecker).VisitExpr(0xc00091ec00, 0x4499060, 0xc000258780)
        /Users/aperson/pkg/mod/github.com/go-critic/go-critic@v0.5.2/checkers/unlambda_checker.go:65 +0x483
github.com/go-critic/go-critic/checkers/internal/astwalk.(*exprWalker).WalkFile.func1(0x4492c60, 0xc000258780, 0x400b901)
        /Users/aperson/pkg/mod/github.com/go-critic/go-critic@v0.5.2/checkers/internal/astwalk/expr_walker.go:23 +0x7f
go/ast.inspector.Visit(0xc00139c820, 0x4492c60, 0xc000258780, 0x44915c0, 0xc00139c820)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/go/ast/walk.go:373 +0x3a
go/ast.Walk(0x44915c0, 0xc00139c820, 0x4492c60, 0xc000258780)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/go/ast/walk.go:52 +0x66
go/ast.walkExprList(0x44915c0, 0xc00139c820, 0xc000258790, 0x1, 0x1)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/go/ast/walk.go:26 +0x9e
go/ast.Walk(0x44915c0, 0xc00139c820, 0x4492620, 0xc0003d3cc0)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/go/ast/walk.go:207 +0x214f
go/ast.walkStmtList(0x44915c0, 0xc00139c820, 0xc000dd5000, 0x2, 0x2)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/go/ast/walk.go:32 +0x9e
go/ast.Walk(0x44915c0, 0xc00139c820, 0x44927a0, 0xc000262ea0)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/go/ast/walk.go:224 +0x1adf
go/ast.Walk(0x44915c0, 0xc00139c820, 0x4492c20, 0xc000262ed0)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/go/ast/walk.go:344 +0xd98
go/ast.Inspect(...)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/go/ast/walk.go:385
github.com/go-critic/go-critic/checkers/internal/astwalk.(*exprWalker).WalkFile(0xc00091ec10, 0xc001108400)
        /Users/aperson/pkg/mod/github.com/go-critic/go-critic@v0.5.2/checkers/internal/astwalk/expr_walker.go:21 +0x12c
github.com/go-critic/go-critic/framework/linter.(*Checker).Check(...)
        /Users/aperson/pkg/mod/github.com/go-critic/go-critic@v0.5.2/framework/linter/lintpack.go:125
github.com/go-critic/go-critic/framework/lintmain/internal/check.(*program).checkFile.func1(0xc0005bcc10, 0xc001108400, 0xc001082000, 0x22, 0x22, 0x1e, 0xc000f35b00)
        /Users/aperson/pkg/mod/github.com/go-critic/go-critic@v0.5.2/framework/lintmain/internal/check/check.go:150 +0xa2
created by github.com/go-critic/go-critic/framework/lintmain/internal/check.(*program).checkFile
        /Users/aperson/pkg/mod/github.com/go-critic/go-critic@v0.5.2/framework/lintmain/internal/check/check.go:132 +0x120

```

--- a/checkers/testdata/unlambda/positive_tests.go
+++ b/checkers/testdata/unlambda/positive_tests.go
@@ -22,6 +22,36 @@ func functionLiterals() {
 	_ = func(x int, y int) int { return add(x, y) }
 }
 
+func variadicInt(xs ...int) int { return 0 }
+
+func variadicTest() {
+	_ = func(x int) int { return variadicInt(x) }
+	_ = func(x int) int { return variadicInt(x, 1) }
+	_ = func(x, y int) int { return variadicInt(x, y) }
+	_ = func(x, y int) int { return variadicInt(x) }
+
+	/*! replace `func(xs ...int) int { return variadicInt(xs...) }` with `variadicInt` */
+	_ = func(xs ...int) int { return variadicInt(xs...) }
+
+	_ = func(x int, ys ...int) int { return variadicInt(1, 2) }
+	_ = func(x int, y int, _ ...int) int { return variadicInt(x, y) }
+}
+
+func variadicInterfaces(x int, y interface{}, ys ...interface{}) int { return 0 }
+
+func TestSomething() {
+	// See #991
+	_ = func(x int, y interface{}, _ ...interface{}) int {
+		return variadicInterfaces(x, y)
+	}
+	_ = func(x int, y interface{}, _ ...interface{}) int {
+		return variadicInterfaces(x, y, 5, "?")
+	}
+
+	/*! replace `func(x int, y interface{}, zs ...interface{}) int { return variadicInterfaces(x, y, zs...) }` with `variadicInterfaces` */
+	_ = func(x int, y interface{}, zs ...interface{}) int { return variadicInterfaces(x, y, zs...) }
+}
+
 type object struct{}
 
 func (object) returnInt(x int) int { return x }

--- a/checkers/unlambda_checker.go
+++ b/checkers/unlambda_checker.go
@@ -2,6 +2,7 @@ package checkers
 
 import (
 	"go/ast"
+	"go/token"
 	"go/types"
 
 	"github.com/go-critic/go-critic/checkers/internal/astwalk"
@@ -61,6 +62,14 @@ func (c *unlambdaChecker) VisitExpr(x ast.Expr) {
 	// Now check that all arguments match the parameters.
 	n := 0
 	for _, params := range fn.Type.Params.List {
+		if _, ok := params.Type.(*ast.Ellipsis); ok {
+			if result.Ellipsis == token.NoPos {
+				return
+			}
+			n++
+			continue
+		}
+
 		for _, id := range params.Names {
 			if !astequal.Expr(id, result.Args[n]) {
 				return

